material/src/test/scala/com/indix/gocd/s3material/plugin/S3PackageMaterialPollerSpec.scala

```diff
@@ -71,6 +71,38 @@ class S3PackageMaterialPollerSpec extends FlatSpec with MockitoSugar with org.sc
+    val result = sut.handle(getRequest(S3PackageMaterialPoller.REQUEST_LATEST_REVISION_SINCE, """
+                                                                                                |{
+                                                                                                |    "repository-configuration": {
+                                                                                                |        "S3_BUCKET": {
+                                                                                                |            "value": "S3 Bucket"
+                                                                                                |        }
+                                                                                                |    },
+                                                                                                |    "package-configuration": {
+                                                                                                |        "PIPELINE_NAME": {
+                                                                                                |            "value": "Pipeline"
+                                                                                                |        },
+                                                                                                |        "STAGE_NAME": {
+                                                                                                |            "value": "Stage"
+                                                                                                |        },
+                                                                                                |        "JOB_NAME": {
+                                                                                                |            "value": "Job"
+                                                                                                |        }
+                                                                                                |    },
+                                                                                                |    "previous-revision": {
+                                                                                                |        "revision": "1.1",
+                                                                                                |        "timestamp": ""
+                                                                                                |    }
+                                                                                                |}
+                                                                                              """.stripMargin))
+    val resultMap = new GsonBuilder().create.fromJson[AnyRef](result.responseBody(), classOf[Any])
+    assert(result.responseCode() == 200)
```

Comment:
I can't understand this. I send a `INTERNAL_ERROR` when handling this particular request from the Plugin side via 'createResponse(DefaultGoPluginApiResponse.INTERNAL_ERROR, revision.toMap())`. However, the response that's received still has 200 status. I checked other tests which asserts some failures, which returns `INTERNAL_ERROR`, they also have assertions expecting a 200 response code. Perhaps, is this how, plugin api response is expected?